### PR TITLE
Support Wasm32 build for RPC-client

### PR DESCRIPTION
**Pull request by XX**
#### Problem

Solana RPC-client does not build for wasm32. 

#### Summary of Changes

Used conditional compilation to switch dependencies for build the RPC-client for wasm32.

Related Issue: #23552

### CODE REVIEW

**Comment by mvines** [comment]
@CriesofCarrots - next week sometime, can you please take a look at this. Overall looks reasonable. 

@XX - My main concern is potential regressions since we aren't building RpcClient for wasm in CI currently. But maybe that's just a matter of adding the rpc client crates to https://github.com/solana-labs/solana/blob/e92f93c29061659c5775cbcd9b7bd48ef235c6c8/ci/test-stable.sh#L217-L228 ?

File: rpc-client/Cargo.toml [hunk 1]
```diff
@@ -26,9 +26,14 @@ solana-rpc-client-api = { path = "../rpc-client-api", version = "=1.15.0" }
 solana-sdk = { path = "../sdk", version = "=1.15.0" }
 solana-transaction-status = { path = "../transaction-status", version = "=1.15.0" }
 solana-version = { path = "../version", version = "=1.15.0" }
-solana-vote-program = { path = "../programs/vote", version = "=1.15.0" }
```
Comment (CriesofCarrots) [inline rpc-client/Cargo.toml:29]: Oops, I think we need to keep solana-vote-program

**Review by CriesofCarrots — COMMENTED** [review]
> Overall looks reasonable
Agreed, looks reasonable. I think we definitely want to add a build to CI, otherwise it will be easy for engs to accidentally break it by adding a dependency in the wrong place or omitting a tag.

@XX , could you please split the part of the changeset that is moving types around to a separate commit to ease review?
Also, could you please run `/scripts/cargo-for-all-lock-files.sh tree` to propagate the Cargo.lock changes?

**Comment by CriesofCarrots** [comment]
@XX , I'm happy to resuscitate this, if you want to address my review comments

**Pull request closed without merging**